### PR TITLE
Replace TryAllowedResult with bool

### PR DESCRIPTION
**Pull request by janisz**
## Description

Replace two value enum with boolean to simplify the code.

## Testing Performed

CI

### CODE REVIEW

**Comment by janisz** [comment]
Current dependencies on/for this PR:
* master
  * **PR #2726** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2726" target="_blank">Graphite</a> 
    * **PR #2722** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2722" target="_blank">Graphite</a> 
      * **PR #2750** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2750" target="_blank">Graphite</a> 
        * **PR #2751** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2751" target="_blank">Graphite</a> 
          * **PR #2752** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2752" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/2752?utm_source=stack-comment).

**Review by rhybrillou — COMMENTED** [review]
small documentation nits in the built in authorizer.
Great work otherwise.

File: central/sac/authorizer/builtin_scoped_authorizer.go [hunk 1]
```diff
@@ -162,23 +162,23 @@ type resourceLevelScopeCheckerCore struct {
 	resource permissions.ResourceMetadata
 }
 
-func (a *resourceLevelScopeCheckerCore) TryAllowed() sac.TryAllowedResult {
+func (a *resourceLevelScopeCheckerCore) Allowed() bool {
```
[the remark this answers is not in the document]
Comment (janisz) [reply central/sac/authorizer/builtin_scoped_authorizer.go:165]: Fixed

File: central/sac/authorizer/builtin_scoped_authorizer.go [hunk 2]
```diff
@@ -77,8 +77,8 @@ type globalScopeChecker struct {
 	trace *observe.AuthzTrace
 }
 
-func (a *globalScopeChecker) TryAllowed() sac.TryAllowedResult {
-	return sac.Deny
+func (a *globalScopeChecker) Allowed() bool {
```
[the remark this answers is not in the document]
Comment (janisz) [reply central/sac/authorizer/builtin_scoped_authorizer.go:80]: Fixed

**Review by rhybrillou — APPROVED** [review]

**Pull request merged**